src/worker_thread.c

```diff
@@ -25,42 +25,48 @@ typedef struct WORKER_THREAD_TAG
     WORKER_FUNC worker_func;
     void* worker_func_context;
     THREAD_HANDLE thread_handle;
-    HANDLE worker_thread_shutdown_event;
-    HANDLE execute_worker_event;
+    volatile_atomic int32_t state;
```

Comment:
>volatile_atomic int32_t state; [](http://example.com/codeflow?start=4&length=30)

if this state is "the same as" `WORKER_THREAD_STATE` then consider having a union over the two. When printing it with PRI_MU_ENUM, use the enum field of the union; when manipulating it with interlocked, use the volatile_atomic member of the union.